client/src/lib/CKANApi/DataStore/tests/Query-test.js

```diff
@@ -1,11 +1,25 @@
 import Query from 'lib/CKANApi/DataStore/Query';
 
 describe('Query', () => {
-  const query = new Query(['Name', 'Date']);
+  let query;
   beforeEach(() => {
-    query.clearFilters();
-    query.clearOrder();
-    query.clearDistinct();
+    query = new Query(['Name', 'Date']);
+  });
+
+  describe('fields setter', () => {
+    it('Only allows arrays', () => {
+      [true, false, 23, {}, { test: 1 }, undefined, null, () => {}].forEach(value => {
+        expect(() => {
+          query.fields = value;
+        }).toThrow('Query.fields must be an Array');
+
+        expect(() => {
+          query.fields = [];
+          query.fields = ['one'];
+          query.fields = ['one', 'two', 'three', 'four'];
+        }).not.toThrow();
```

Comment:
This doesn't look like it needs to be inside the loop

Reply:
Yep. My mistake